test/ecto/changeset_test.exs

```diff
@@ -1061,6 +1061,26 @@ defmodule Ecto.ChangesetTest do
     assert changeset.errors == [topics: {"can't be blank", [validation: :required]}]
   end
 
+  test "filter_required/2" do
+    required_fields =
+      changeset(%{"body" => "something"})
+      |> filter_required([:title, :body])
+
+    assert [:body] == required_fields
+
+    required_fields =
+      changeset(%{"body" => "something", "title" => "Title"})
+      |> filter_required([:title, :body])
+
+    assert [:title, :body] == required_fields
+
+    required_fields =
+      changeset(%{})
+      |> filter_required([:title, :body])
+
+    assert [] == required_fields
+  end
```

Comment:
consider adding a test that raises when someone passed a field that doesn't exist.